src/averageAges.js

```diff
@@ -55,10 +75,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = onlyWithSon
+    ? people.filter(person => person.sex === 'm' && person.mother !== null)
+    : people.filter(person => person.mother !== null);
```

Comment:
```suggestion
   const peopleWithMother = people.filter(person => onlyWithSon
      ? mothers.find(mother => mother.name === person.mother)
        && person.sex === 'm'
      : mothers.find(mother => mother.name === person.mother);
  );
```

Reply:
The fact that the mother is not null does not guarantee that we will be able to find it in this array